### ISSUE TITLE
How can I try this vlm2vec custom image datasets with class labels?

### ISSUE THREAD

**Issue by chandrabhuma** [TIGER-AI-Lab/VLM2Vec#34]
Can I get a single embedding for image and class labels so that I will be able to apply any classifier like svm? Is it fair to compare with clip models when the train dataset is completely different as clip models were trained LAION and WebLI,, DFN etc.

**Comment by XMHZZ2018** [comment]
Hi @chandrabhuma , thank you for your interest in our work! Yes, it totally makes sense to use our model to encode both images and class labels. You might also consider trying zero-shot similarity search for classification, but it still makes sense to fine-tune a separate classifier like an SVM. I think you can definitely compare it with the CLIP model, even though it uses different training data. After all, existing VLMs often rely on different sources of pretraining data, and it's nearly impossible to ensure they're using the exact same training data.
